### PR TITLE
@mzikherman - Fillwidth component

### PR DESCRIPTION
**Pull request by broskoski**
After a few false starts I'm starting to understand the react / graphql version of replicating @craigspaeth's jQuery filldwidth plugin. This will definitely be less generic than that, the reason being that we can make use of the `aspect_ratio` attribute that gets returned on artwork images (which makes calculating a row a bit easier, especially not having to wait for images to load.

This doesn't really do much atm, just getting some code up here.

### CODE REVIEW

**Comment by broskoski** [comment]
Hm, looks like `Dimensions` screws with the type-check...that will be fun to figure out.

**Comment by broskoski** [comment]
Looking like this so far:
(6 artworks)
screenshot 2017-05-17 12 10 20

(10 artworks)
screenshot 2017-05-17 12 09 36

File: tsconfig.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
         "allowSyntheticDefaultImports": true,
         "baseUrl": "src",
         "outDir": "dist",
-        "noUnusedLocals": true
+        "noUnusedLocals": false
```
Comment (alloy) [inline tsconfig.json]: Why is this change needed?

Comment (broskoski) [inline tsconfig.json]: We already check for it in esLint and as far as I can tell there is no way to disable on a per-line basis (https://github.com/artsy/reaction-force/pull/162/files#diff-b7c8554bab28defe2bb4f7c32a1e2d33R68)

Comment (alloy) [inline tsconfig.json]: We don’t use eslint but tslint, which I iirc doesn’t catch e.g. unused imports.

File: src/components/artwork/fillwidth.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    })
+
+    // If the total width difference is less than 0, it is larger than the container
+    // so we need to scale down. If not, scale up.
+    let dir = this.widthDiff(dimensions) < 0 ? -1 : 1
+
+    // Keep looping until we get an acceptable width difference
+    const list = new Array(999)
+
+    for (let i of list) { // eslint-disable-line no-unused-vars
```
Comment (alloy) [inline src/components/artwork/fillwidth.tsx]: Is the intention here to just loop indefinitely until `break` is called? If so, something like the following will get rid of the unused array and unused variable:

```js
let proceed = true
while (proceed) {
  // ...
}
```

Comment (broskoski) [inline src/components/artwork/fillwidth.tsx]: 👍

File: src/components/artwork/fillwidth.tsx [hunk 3]
```diff
@@ -63,9 +63,7 @@ export class Fillwidth extends React.Component<Props, null> {
     let dir = this.widthDiff(dimensions) < 0 ? -1 : 1
 
     // Keep looping until we get an acceptable width difference
-    const list = new Array(999)
-
-    for (let i of list) { // eslint-disable-line no-unused-vars
+    while (true) {
```
Comment (alloy) [inline src/components/artwork/fillwidth.tsx]: Even better! I wasn’t sure what the rules were for nested loops (i.e. the `for` inside this `while`) 👍

**Comment by broskoski** [comment]
This is ready to go, what say y'all?

**Comment by alloy** [comment]
Let’s 👍
